Process/line_warpers.py

```diff
@@ -0,0 +1,92 @@
+"""Module for and generating altering line segments to assist processing and defining puzzle pieces."""
+
+import numpy as np
+import math
+
+def rotate(origin, point, angle):
+    """
+    Rotate a point counterclockwise by a given angle around a given origin.
+    From: https://stackoverflow.com/questions/34372480/rotate-point-about-another-point-in-degrees-python
+
+    The angle should be given in radians.
+    """
+    ox, oy = origin
+    px, py = point
+
+    qx = ox + math.cos(angle) * (px - ox) - math.sin(angle) * (py - oy)
+    qy = oy + math.sin(angle) * (px - ox) + math.cos(angle) * (py - oy)
+    return qx, qy
+
+def midpoint(p1, p2):
```

Comment:
I would name this something more descriptive, like `compute_midpoint` (or similar)